comp/process/status/statusimpl/status_test.go

```diff
@@ -7,15 +7,214 @@ package statusimpl
 
 import (
 	"bytes"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func TestStatusOut(t *testing.T) {
-	provides := newStatus()
+var jsonResponse = `{
+	"core": {
+		"build_arch": "arm64",
+		"config": {
+			"log_level": "info"
+		},
+		"go_version": "go1.21.5",
+		"metadata": {
+			"agent-flavor": "process_agent",
+			"host-tags": {
+				"system": []
+			},
+			"install-method": {
+				"installer_version": null,
+				"tool": null,
+				"tool_version": "undefined"
+			},
+			"logs": {
+				"auto_multi_line_detection_enabled": false,
+				"transport": ""
+			},
+			"meta": {
+				"ec2-hostname": "",
+				"host_aliases": [],
+				"hostname": "COMP-VQHPF4W6GY",
+				"instance-id": "",
+				"socket-fqdn": "COMP-VQHPF4W6GY",
+				"socket-hostname": "COMP-VQHPF4W6GY",
+				"timezones": [
+					"CET"
+				]
+			},
+			"network": null,
+			"os": "darwin",
+			"otlp": {
+				"enabled": false
+			},
+			"proxy-info": {
+				"no-proxy-nonexact-match": false,
+				"no-proxy-nonexact-match-explicitly-set": false,
+				"proxy-behavior-changed": false
+			},
+			"python": "n/a",
+			"systemStats": {
+				"cpuCores": 10,
+				"fbsdV": [
+					"",
+					"",
+					""
+				],
+				"macV": [
+					"14.2.1",
+					[
+						"",
+						"",
+						""
+					],
+					"arm64"
+				],
+				"machine": "arm64",
+				"nixV": [
+					"",
+					"",
+					""
+				],
+				"platform": "darwin",
+				"processor": "Apple M1 Max",
+				"pythonV": "n/a",
+				"winV": [
+					"",
+					"",
+					""
+				]
+			}
+		},
+		"version": "7.51.0-rc.1+git.416.0d1edc1"
+	},
+	"date": 1706892483712089000,
+	"expvars": {
+		"connections_queue_bytes": 0,
+		"connections_queue_size": 0,
+		"container_count": 0,
+		"container_id": "",
+		"docker_socket": "/var/run/docker.sock",
+		"drop_check_payloads": [],
+		"enabled_checks": [
+			"process",
+			"rtprocess"
+		],
+		"endpoints": {
+			"https://process.datadoghq.eu": [
+				"72724"
+			]
+		},
+		"event_queue_bytes": 0,
+		"event_queue_size": 0,
+		"language_detection_enabled": false,
+		"last_collect_time": "2024-02-02 17:47:57",
+		"log_file": "",
+		"memstats": {
+			"alloc": 30387880
+		},
+		"pid": 72211,
+		"pod_queue_bytes": 0,
+		"pod_queue_size": 0,
+		"process_count": 757,
+		"process_queue_bytes": 0,
+		"process_queue_size": 0,
+		"proxy_url": "",
+		"rtprocess_queue_bytes": 0,
+		"rtprocess_queue_size": 0,
+		"system_probe_process_module_enabled": false,
+		"uptime": 18,
+		"uptime_nano": 1706892464835469000,
+		"version": {
+			"BuildDate": "",
+			"GitBranch": "",
+			"GitCommit": "",
+			"GoVersion": "",
+			"Version": ""
+		},
+		"workloadmeta_extractor_cache_size": 0,
+		"workloadmeta_extractor_diffs_dropped": 0,
+		"workloadmeta_extractor_stale_diffs": 0
+	}
+}
+`
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You could leverage json formatting storing this into a fixture and importing with go embed https://pkg.go.dev/embed

Reply:
Great idea 😄

Reply:
Done https://github.com/DataDog/datadog-agent/pull/22527/commits/561e2768d7fdff939412c46f04d31f0de50433ea